Algorithm/Portfolio/PortfolioConstructionModel.cs

```diff
@@ -43,5 +46,39 @@ public virtual IEnumerable<IPortfolioTarget> CreateTargets(QCAlgorithm algorithm
         public virtual void OnSecuritiesChanged(QCAlgorithm algorithm, SecurityChanges changes)
         {
         }
+
+        /// <summary>
+        /// Helper class that can be used by the different <see cref="IPortfolioConstructionModel"/>
+        /// implementations to filter <see cref="Insight"/> instances with an invalid
+        /// <see cref="Insight.Magnitude"/> value based on the <see cref="IAlgorithmSettings"/>
+        /// </summary>
+        /// <param name="algorithm">The algorithm instance</param>
+        /// <param name="insights">The insight collection to filter</param>
+        /// <returns>Returns a new array of insights removing invalid ones</returns>
+        public static Insight[] FilterInvalidInsightMagnitude(QCAlgorithm algorithm, Insight[] insights)
+        {
+            var result = insights.Where(insight =>
+            {
+                if (!insight.Magnitude.HasValue || insight.Magnitude == 0)
+                {
+                    return true;
+                }
+
+                var absoluteMagnitude = Math.Abs(insight.Magnitude.Value);
+                if (absoluteMagnitude > (double)algorithm.Settings.MaxAbsolutePortfolioTargetPercentage
+                    || absoluteMagnitude < (double)algorithm.Settings.MinAbsolutePortfolioTargetPercentage)
+                {
+                    algorithm.Error("PortfolioConstructionModel.FilterInvalidInsightMagnitude():" +
+                        $"The insight target Magnitude: {insight.Magnitude}, will not comply with the current " +
```

Comment:
Should be `does not comply` or is it intentional?

Reply:
This one is tricky 😄 because the `Insight.Magnitude` **might** not directly match (==) the resulting (**not yet created**) `IPortfolioTarget.Percentage` because it depends on the implementation of the `IPortfolioConstructionModel`. So with the message I'm trying to point out that the potential resulting `IPortfolioTarget.Percentage` won't comply with the settings